src/components/ReservationForm.tsx

```diff
@@ -0,0 +1,490 @@
+import React, {
+  FunctionComponent,
+  useEffect,
+  useContext,
+  useState,
+} from "react";
+import TextField from "@material-ui/core/TextField";
+import Radio from "@material-ui/core/Radio";
+import RadioGroup from "@material-ui/core/RadioGroup";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import FormControl from "@material-ui/core/FormControl";
+import FormLabel from "@material-ui/core/FormLabel";
+import { CalendarUIProps, CalendarAction } from "../calendar/types";
+import {
+  IconButton,
+  makeStyles,
+  Dialog,
+  Toolbar,
+  Typography,
+  DialogContent,
+  Button,
+} from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+import Select from "./Select";
+import { AuthContext } from "./AuthContext";
+import { makeTransition } from "./Transition";
+import UserGroup from "../resources/UserGroup";
+import { Formik } from "formik";
+import * as Yup from "yup";
+import GearForm from "./GearForm";
+import QuantityList from "./QuantityList";
+import { ResourceKey } from "../resources/types";
+import Project from "../resources/Project";
+import Equipment from "../resources/Equipment";
+
+const useStyles = makeStyles(() => ({
+  guests: {
+    display: "none",
+  },
+  notes: {
+    display: "none",
+  },
+  gearList: {
+    display: "none",
+  },
+  list: {
+    display: "flex",
+    justifyContent: "flex-start",
+    alignContent: "row",
+  },
+}));
+
+const transition = makeTransition("left");
+
+const ReservationForm: FunctionComponent<CalendarUIProps> = ({
+  dispatch,
+  state,
+}) => {
+  // Get values from App
+  const { user } = useContext(AuthContext);
+  const groups = state.resources[ResourceKey.Groups] as UserGroup[];
+  const projects = state.resources[ResourceKey.Projects] as Project[];
+  const gear = state.resources[ResourceKey.Equipment] as Equipment[];
+
+  console.log(gear);
+  // Function to convert Gear Array to Quantized Gear Array
+  function quantizeGear(gear: Equipment[]): Equipment[] {
+    const tempArray: Equipment[] = [];
+    gear.forEach((item) => {
+      item.quantity = gear.filter(
+        (element) => element.title === item.title
+      ).length;
+      const index = tempArray.findIndex(
+        (element) => element.title === item.title
+      );
+      if (index === -1) {
+        tempArray.push(item);
+      }
+    });
+    return tempArray;
+  }
```

Comment:
i've implemented the first suggestion in 5aebe0c, I tried the second and the app freezes when opening the event detail. I'll take a second look at this, since it is a very clever way to do the quantization, but I'm going to move on for now